examples/fuse-fd-mount-point/mounthelper.go

```diff
@@ -0,0 +1,112 @@
+// An example script showing usage of FUSE file descriptor as a mount point.
+//
+// Example usage:
+// $ go build mounthelper.go
+// $ sudo /sbin/setcap 'cap_sys_admin=ep' ./mounthelper # `mount` syscall requires `CAP_SYS_ADMIN`, alternatively, `mounthelper` can be run as root
+// $ ./mounthelper -mountpoint /tmp/mountpoint -bucket bucketname
+// $ # Mountpoint mounted at /tmp/mountpoint until `mounthelper` is terminated with ctrl+c.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+)
+
+var mountPoint = flag.String("mountpoint", "", "Path to mount the filesystem at")
+var bucket = flag.String("bucket", "", "S3 Bucket to mount")
+
+func main() {
+	flag.Parse()
+
+	// `os.MkdirAll` will return `nil` if `mountPoint` is an already existing directory.
+	if err := os.MkdirAll(*mountPoint, 0644); err != nil {
+		log.Panicf("Failed to create target mount point %s: %v\n", *mountPoint, err)
+	}
+
+	// 1. Open FUSE device
+	const USE_DEFAULT_PERM = 0
+	fd, err := syscall.Open("/dev/fuse", os.O_RDWR, USE_DEFAULT_PERM)
+	if err != nil {
+		log.Panicf("Failed to open /dev/fuse: %v\n", err)
+	}
+
+	// Ensure to close the file descriptor in case of an error, if there are no errors,
+	// the file descriptor will be closed and `closeFd` will set to `false`.
+	closeFd := true
+	defer func() {
+		if closeFd {
+			err := syscall.Close(fd)
+			if err != nil {
+				log.Panicf("Failed to close fd on parent: %v\n", err)
+			}
+		}
+	}()
+
+	var stat syscall.Stat_t
+	err = syscall.Stat(*mountPoint, &stat)
+	if err != nil {
+		log.Panicf("Failed to stat mount point %s: %v\n", *mountPoint, err)
+	}
+
+	// 2. Perform `mount` syscall
+	// Mountpoint enables and recommends these mount options and flags by default
```

Comment:
Mountpoint sets `noatime` and `default_permissions` [by default](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/cli.rs#L527-L531), and fuser just passes them to kernel as far as I can see. So there is no logic on userspace rather than passing those while performing `mount` syscall.